__tests__/app.test.js

```diff
@@ -143,6 +166,70 @@ describe("/api/articles", () => {
         });
     });
   });
+
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("Responds with the posted comment", () => {
+      return request(app)
+        .post("/api/articles/3/comments")
+        .send({
+          username: "butter_bridge",
+          body: "I love napping in the sun!",
+        })
+        .expect(201)
+        .then(({ body }) => {
+          const { comment } = body;
+          expect(comment).toMatchObject({
+            comment_id: 19,
+            body: "I love napping in the sun!",
+            article_id: 3,
+            author: "butter_bridge",
+            votes: 0,
+          });
+
+          expect(comment.created_at).toEqual(expect.any(String));
+        });
+    });
+
+    test("POST 400: Responds with an adequate status and error message when provided with an incomplete body", () => {
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send({ username: "butter_bridge" })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.message).toBe("Bad request.");
+        });
+    });
+
+    test("POST 400: Responds with an adequate status and error message when the data type of the object's property values are not correct", () => {
+      return request(app)
+        .post("/api/articles/2/comments")
+        .send({ username: 200, body: false })
+        .expect(400)
```

Comment:
these tests are great - what if the username was "Pippa" - what error would we get then?

Reply:
Would it be a 400? It would trigger an psql error.